### ISSUE TITLE
Array parsing does not work as expected

### ISSUE THREAD

**Issue by adrianignat13** [jdevillard/JmesPath.Net#57]
Give this input:

```
{
              "payload": {
                "partId": "",
                "mimeType": "multipart/alternative",
                "filename": "",
                "headers": [
                  {
                    "name": "Delivered-To",
                    "value": "something@uipath.rocks"
                  },
                  {
                    "name": "To",
                    "value": "qa@uipath.com"
                  }
                ]
              }
            }
```
and this expression
`payload.headers[?not_null(name,'')=='Delivered-To'].value[?contains(not_null(@,''), '@uipath.rocks')]`

the library returns an empty array [] but it should return 
["something@uipath.rocks"]

Validated it with: https://jsoneditoronline.org/#left=local.xahute

**Comment by springcomp** [comment]
@adrianignat13 Thank you for your feedback.

It appears that the expression you shared is not actually valid.
Here is a simplified example that manifests the issue.

Consider the input JSON document:

```json
[
  {"value": "something@uitpath.rocks"}
]
```

The expression that manifests your issue would be: `` [].value[?`true`] ``.
An alternate expression to return the first item of the array would look like: `` [].value[0] ``.

Bot of which return an empty array as per the specification.

The reason for this is that the last _filter-expression_ `` [?`true`] `` applies to a _projection_. Which means that every item in the preceding array is take from the array and applied to the _filter-expression_.

The correct expression you actually want is: `` [].value |[?`true`] ``.
Much like the correct expression to return the first item in the array is: `` [].value |[0] ``.

Using a `|` stops the projection and has the resulting expression (either the `` [?`true`] `` _filter-expression_ or `[0]` _index-expression_) applied to the array, instead of multiple-times on each item from the array.

# TL; DR

The correct expression you want is:

`` payload.headers[?not_null(name,'')=='Delivered-To'].value | [?contains(not_null(@,''), '@uipath.rocks')] ``
